File: PrintService/Modulos/FormatoFechas.vb
Public Class FormatoFechas

        Public Shared Function fFecha(ByVal pFecha As Date, Optional ByVal Con_Hora As Boolean = False) As String
            Try
                fFecha = ""
                If IsDate(pFecha) Then
                    If Con_Hora Then
                        Dim Hora As String = String.Format("{0}:{1}:{2}", pFecha.Hour, Right("00" & pFecha.Minute, 2), Right("00" & pFecha.Second, 2))
                        fFecha = String.Format("'{0}{1}{2} {3}'", CStr(pFecha.Year), Right("00" & pFecha.Month, 2), Right("00" & pFecha.Day, 2), Hora)
                    Else
                        fFecha = String.Format("'{0}{1}{2}'", CStr(pFecha.Year), Right("00" & pFecha.Month, 2), Right("00" & pFecha.Day, 2))
                    End If
                End If
            Catch ex As Exception
                Throw New Exception(ex.Message)
            End Try
        End Function

        Public Shared Function tFecha(ByVal pFecha As Date) As String
            tFecha = ""
            Try
                If IsDate(pFecha) Then
                    tFecha = CStr(pFecha.Year) & Right("00" & pFecha.Month, 2) & Right("00" & pFecha.Day, 2)
                End If
            Catch ex As Exception
                Throw New Exception(ex.Message)
            End Try
        End Function

        Public Shared Function sFecha(ByVal pFecha As Date) As String
            Try
                sFecha = ""
                If IsDate(pFecha) Then
                    sFecha = String.Format("{0}/{1}/{2}", Right(String.Format("00{0}", pFecha.Day), 2), Right("00" & pFecha.Month, 2), CStr(pFecha.Year))
                End If
            Catch ex As Exception
                Throw New Exception(ex.Message)
            End Try
        End Function

        Public Shared Function sHora(ByVal pFecha As Date) As String
            Try
                sHora = ""
                If IsDate(pFecha) Then
                    sHora = String.Format("{0}:{1}:{2}", Right("00" & pFecha.Hour, 2), Right("00" & pFecha.Minute, 2), CStr(pFecha.Second))
                End If
            Catch ex As Exception
                Throw New Exception(ex.Message)
            End Try
        End Function

        Public Shared Function fFechaHora(ByVal pFecha As Date) As String
            fFechaHora = "'01-Jan-1900 00:00:00'"
            If IsDate(pFecha) Then
                fFechaHora = String.Format("'{0}{1}{2} ", CStr(pFecha.Year), Right("00" & pFecha.Month, 2), Right("00" & pFecha.Day, 2))
                fFechaHora = String.Format("{0}{1}:{2}:{3}'", fFechaHora, CStr(pFecha.Hour), CStr(pFecha.Minute), CStr(pFecha.Second))
            End If
        End Function

    End Class

File: WMS.AppGlobalCore/FormatoFechas.cs
using Microsoft.VisualBasic;
public class FormatoFechas
{
    public static string fFecha(DateTime pFecha, bool Con_Hora = false)
    {

        string vfFecha = "";

        try
        {

            if (Information.IsDate(pFecha))
            {
                if (Con_Hora)
                {
                    string Hora = string.Format("{0}:{1}:{2}", pFecha.Hour, Strings.Right("00" + pFecha.Minute, 2), Strings.Right("00" + pFecha.Second, 2));
                    vfFecha = string.Format("'{0}{1}{2} {3}'", pFecha.Year.ToString(), Strings.Right("00" + pFecha.Month, 2), Strings.Right("00" + pFecha.Day, 2), Hora);
                }
                else
                    vfFecha = string.Format("'{0}{1}{2}'", pFecha.Year.ToString(), Strings.Right("00" + pFecha.Month, 2), Strings.Right("00" + pFecha.Day, 2));
            }
        }
        catch (Exception ex)
        {
            throw new Exception(ex.Message);
        }

        return vfFecha;
    }

    public static string tFecha(DateTime pFecha)
    {
        string vfFecha = "";
        try
        {
            if (Information.IsDate(pFecha))
                vfFecha = Convert.ToString(pFecha.Year) + Strings.Right("00" + pFecha.Month, 2) + Strings.Right("00" + pFecha.Day, 2);
        }
        catch (Exception ex)
        {
            throw new Exception(ex.Message);
        }
        return vfFecha;
    }

    public static string sFecha(DateTime pFecha)
    {
        string vfFecha = "";
        try
        {
            if (Information.IsDate(pFecha))
                vfFecha = string.Format("{0}/{1}/{2}", Strings.Right(string.Format("00{0}", pFecha.Day), 2), Strings.Right("00" + pFecha.Month, 2), pFecha.Year.ToString());
        }
        catch (Exception ex)
        {
            throw new Exception(ex.Message);
        }
        return vfFecha;
    }

    public static string sFechaM(DateTime pFecha)
    {
        string vfFecha = "";
        try
        {
            if (Information.IsDate(pFecha))
                vfFecha = string.Format("{0}/{1}/{2}", Strings.Right(string.Format("00{0}", pFecha.Month), 2), Strings.Right("00" + pFecha.Day, 2), pFecha.Year.ToString());
        }
        catch (Exception ex)
        {
            throw new Exception(ex.Message);
        }
        return vfFecha;
    }

    public static string sHora(DateTime pFecha)
    {
        string vsHora = "";
        try
        {
            if (Information.IsDate(pFecha))
                vsHora = string.Format("{0}:{1}:{2}", Strings.Right("00" + pFecha.Hour, 2), Strings.Right("00" + pFecha.Minute, 2), pFecha.Second.ToString());
        }
        catch (Exception ex)
        {
            throw new Exception(ex.Message);
        }
        return vsHora;
    }

    public static string fFechaHora(DateTime pFecha)
    {
        string vfFecha = "'01-Jan-1900 00:00:00'";
        if (Information.IsDate(pFecha))
        {
            vfFecha = string.Format("'{0}{1}{2} ", pFecha.Year.ToString(), Strings.Right("00" + pFecha.Month, 2), Strings.Right("00" + pFecha.Day, 2));
            vfFecha = string.Format("{0}{1}:{2}:{3}'", vfFecha, pFecha.Hour.ToString(), pFecha.Minute.ToString(), pFecha.Second.ToString());
        }
        return vfFecha;
    }

    public static string fFechaHora_Time(DateTime pFecha)
    {
        string vfFecha = "'01-Jan-1900 00:00:00'";
        if (Information.IsDate(pFecha))
        {
            vfFecha = string.Format("'{0}{1}{2} ", pFecha.Year.ToString(), Strings.Right("00" + pFecha.Month, 2), Strings.Right("00" + pFecha.Day, 2));
            vfFecha = string.Format("{0}{1}:{2}:{3}'", vfFecha, pFecha.Hour.ToString(), pFecha.Minute.ToString(), pFecha.Second.ToString());
        }
        return vfFecha;
    }

    public static DateTime sFecha_To_Date(string pFecha)
    {
        DateTime vsFecha_To_Date = new DateTime(1900, 1, 1);
        try
        {

            int vAño = int.Parse(pFecha.Substring(0, 4).ToString());
            int vMes = int.Parse(pFecha.Substring(4, 2).ToString());
            int vDia = int.Parse(pFecha.Substring(6, 2).ToString());

            vsFecha_To_Date = new DateTime(vAño, vMes, vDia);
        }
        catch (Exception)
        {
            throw;
        }
        return vsFecha_To_Date;
    }
}